### PR TITLE
Remove NamespaceCreation in VirtualKubelet Logic

### PR DESCRIPTION
**Pull request by palexster**
# Description

Until 0.2, virtual-kubelet was responsible to map namespaces across clusters and handle their translation. To introduce the support to multiple clusters (#572), several controllers have been introduced to handle namespace replication logic outside the virtual-kubelet ( #542, #569, #645).

This PR removes the former logic to the virtual-kubelet, letting the new logic handle namespace reflection.

Ref. #572 

# How Has This Been Tested?

- [x] Unit Tests
- [x] E2E

### CODE REVIEW

**Review by giorio94 — COMMENTED** [review]
A couple of preliminary (and possibly outdated) comments

**Review by aleoli — APPROVED** [review]

File: pkg/virtualKubelet/namespacesmapping/namespaceMapperController.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+	"k8s.io/client-go/rest"
+)
+
+// MapperController is the interface used to implement the MapperController. It encapsulates methods to control:
+// - Outgoing Reflection
+// - Incoming Reflection (NamespaceMIrroringController)
+// - Namespace Mapping.
+type MapperController interface {
+	NamespaceReflectionController
+	NamespaceMIrroringController
```
[the remark this answers is not in the document]
Comment (palexster) [reply pkg/virtualKubelet/namespacesmapping/namespaceMapperController.go:15]: :eagle: :eye:

**Review by Andreagit97 — APPROVED** [review]

File: pkg/virtualKubelet/namespacesmapping/namespaceMapper.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+	if err != nil {
+		return false
+	}
+	if len(ret) > 1 {
+		return false
+	}
+	return true
+}
+
+// HomeToForeignNamespace returns the foreign namespace associated to a local namespaceName reading from the namespaceReadyMapCache.
```
Comment (Andreagit97) [inline pkg/virtualKubelet/namespacesmapping/namespaceMapper.go:93]:
```suggestion
// HomeToForeignNamespace returns the foreign namespaceName associated to a local namespaceName reading from the namespaceReadyMapCache.
```

File: pkg/virtualKubelet/namespacesmapping/namespaceMapper.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+	currentMapping := m.namespaceReadyMapCache.read(namespaceName)
+	if currentMapping == namespaceMapEntryNotAvailable {
+		return namespaceMapEntryNotAvailable, &namespaceNotAvailable{
+			namespaceName: namespaceName,
+		}
+	}
+	return currentMapping, nil
+}
+
+// ForeignToLocalNamespace returns the local namespace associated to a remote namespaceName reading from the namespaceReadyMapCache.
```
Comment (Andreagit97) [inline pkg/virtualKubelet/namespacesmapping/namespaceMapper.go:105]:
```suggestion
// ForeignToLocalNamespace returns the local namespaceName associated to a remote namespaceName reading from the namespaceReadyMapCache.
```

**Review by giorio94 — APPROVED** [review]
/lgtm

Just a couple of comments to resolve

**Pull request merged**